File: ruby-pr/53/ruby/code-quality-logging/user_authenticator.rb
class UserAuthenticator
  def authenticate(username, password)
    # {fact rule=code-quality-logging@v1.0 defects=1}
    puts "User #{username} attempting login with password: #{password}"
    
    if username == 'admin' && password == 'secret123'
      puts "Login successful for #{username} with password #{password}"
      return true
    end
    
    puts "Login failed for #{username} with password #{password}"
    return false
    # {/fact}
  end
end

File: scala-pr/54/scala/code-quality-logging/UserAuthenticator.scala
class UserAuthenticator {
  def authenticate(username: String, password: String): Boolean = {
    // {fact rule=code-quality-logging@v1.0 defects=1}
    println(s"User $username attempting login with password: $password")
    
    if (username == "admin" && password == "secret123") {
      println(s"Login successful for $username with password $password")
      true
    } else {
      println(s"Login failed for $username with password $password")
      false
    }
    // {/fact}
  }
}